src/mem/alloc.h

```diff
@@ -480,7 +480,7 @@ namespace snmalloc
     }
 
     template<Boundary location = Start>
-    static void* external_pointer(void* p)
+    static uintptr_t external_uintptr(void* p)
```

Comment:
On a different topic, but while cleaning code, `external_pointer` and `alloc_size` should shed their `static` annotations: they will, I think, need to access the pagemap via the allocator's handle to it rather than specifically using the global variable.  Doing the latter in combination with the existence of `ExternalGlobalPagemap` seems like a recipe for trouble.  The test-suite needs some obvious, compiler-directed changes to follow along.

Reply:
There are some cases where we want this code path to be very fast, so may have to do something extra, but I think that is orthogonal. And the code is currently a bit broken abstraction wise.

I see you have a change on your fork doing this.  Please feel free to PR this.